### DISCUSSION TITLE
Calculating an age from a DOB

### DISCUSSION THREAD

**Discussion by PaulMcF1987** [Budibase/budibase#7234] [Help] [answered]
Unsure what I am doing wrong here, Im thinking that it may be because the Date column automatically adds the time but not 100% sure.

I am trying to use a `paragraph` field to display a persons age, based on the Date of Birth that they have entered ion the form.

I have tried a few JS functions, essentially

`x = $(repeater.table.date) - date.now()`

I keep getting a JS error, not sure if the time has something to do with this?

I have tried using handlebars as well but no joy.

Any help would be much appreciated.

**Comment by andz-bb** [comment, marked as answer]
Hi, the `$(repeater.table.date)` will be coming through as a string rather than a Date object. You could parse that string into a new Date object and then compare it with a current Date object to calculate the age. There might be a better way to do it but this example worked for me:
![testing age calculation](https://user-images.githubusercontent.com/110921612/184397319-379dfbfb-4812-4097-931d-de125d90a4d1.png)

Here's the snippet of code I used with your date variable in place, hopefully that or something similar will work for you!

```
const dob = new Date($("repeater.table.date"));
const birthYear = dob.getFullYear();
const currentYear = new Date().getFullYear();

const age = currentYear - birthYear;

return age;
```

Let me know if this doesn't work on your end and I'll be happy to have another look. Thanks 😊

**Reply by melohagan** [reply]
You can also have a look at some StackOverflow examples if you need more precision: https://stackoverflow.com/questions/10008050/get-age-from-birthdate

But this is a great example!
